renderdoc/driver/d3d12/d3d12_counters.cpp

```diff
@@ -251,83 +264,159 @@ vector<CounterResult> D3D12Replay::FetchCounters(const vector<uint32_t> &counter
     return ret;
   }
 
-  D3D12_QUERY_HEAP_DESC queryDesc;
-  queryDesc.Count = maxEID * 2;
-  queryDesc.NodeMask = 1;
-  queryDesc.Type = D3D12_QUERY_HEAP_TYPE_TIMESTAMP;
-  ID3D12QueryHeap *queryHeap = NULL;
-  hr = m_pDevice->CreateQueryHeap(&queryDesc, __uuidof(queryHeap), (void **)&queryHeap);
+  D3D12_QUERY_HEAP_DESC timerQueryDesc;
+  timerQueryDesc.Count = maxEID * 2;
+  timerQueryDesc.NodeMask = 1;
+  timerQueryDesc.Type = D3D12_QUERY_HEAP_TYPE_TIMESTAMP;
+  ID3D12QueryHeap *timerQueryHeap = NULL;
+  hr = m_pDevice->CreateQueryHeap(&timerQueryDesc, __uuidof(timerQueryHeap),
+                                  (void **)&timerQueryHeap);
+  if(FAILED(hr))
+  {
+    RDCERR("Failed to create timer query heap %08x", hr);
+    return ret;
+  }
+
+  D3D12_QUERY_HEAP_DESC pipestatsQueryDesc;
+  pipestatsQueryDesc.Count = maxEID;
+  pipestatsQueryDesc.NodeMask = 1;
+  pipestatsQueryDesc.Type = D3D12_QUERY_HEAP_TYPE_PIPELINE_STATISTICS;
+  ID3D12QueryHeap *pipestatsQueryHeap = NULL;
+  hr = m_pDevice->CreateQueryHeap(&pipestatsQueryDesc, __uuidof(pipestatsQueryHeap),
+                                  (void **)&pipestatsQueryHeap);
+  if(FAILED(hr))
+  {
+    RDCERR("Failed to create pipeline statistics query heap %08x", hr);
+    return ret;
+  }
+
+  D3D12_QUERY_HEAP_DESC occlusionQueryDesc;
+  occlusionQueryDesc.Count = maxEID;
+  occlusionQueryDesc.NodeMask = 1;
+  occlusionQueryDesc.Type = D3D12_QUERY_HEAP_TYPE_OCCLUSION;
+  ID3D12QueryHeap *occlusionQueryHeap = NULL;
+  hr = m_pDevice->CreateQueryHeap(&occlusionQueryDesc, __uuidof(occlusionQueryHeap),
+                                  (void **)&occlusionQueryHeap);
   if(FAILED(hr))
   {
-    RDCERR("Failed to create query heap %08x", hr);
+    RDCERR("Failed to create occlusion query heap %08x", hr);
     return ret;
   }
 
-  m_pDevice->SetStablePowerState(TRUE);
+  // Only supported with developer mode drivers!!!
+  hr = m_pDevice->SetStablePowerState(TRUE);
+  if(FAILED(hr))
+    MessageBoxW(NULL, (LPCWSTR)L"D3D12 counters require Win10 developer mode enabled: Settings > Update & Security > For Developers > Developer Mode", (LPCWSTR)L"D3D12 Counters Error", MB_ICONWARNING | MB_OK);
```

Comment:
These casts shouldn't be necessary and just add clutter. I'd say just use `MessageBoxA` and drop the casts - there's no need for the wide version when you're using ASCII strings anyway.